addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1781,31 +1809,54 @@ const ListUserValueWidget = UserValueWidget.extend({
+
+        if (this.createWidget) {
+            const selectedIds = currentValues.map(({ id }) => id)
+                .filter(id => typeof id === 'number');
+            this.createWidget.options.domain = ['!', ['id', 'in', selectedIds]];
+            this.createWidget.setValue('');
+            this.createWidget.inputEl.value = '';
+            $(this.createWidget.inputEl).trigger('input');
         } else {
-            // TODO use a real select widget ?
-            this.addItemButton = document.createElement('we-select');
-            this.addItemButton.classList.add('o_we_user_value_widget');
-            const divEl = document.createElement('div');
-            this.addItemButton.appendChild(divEl);
-            const togglerEl = document.createElement('we-toggler');
-            togglerEl.textContent = this.addItemTitle;
-            divEl.appendChild(togglerEl);
-            this.selectMenuEl = document.createElement('we-selection-items');
-            divEl.appendChild(this.selectMenuEl);
-            currentValues.forEach(val => {
-                const record = this.records.find(rec => rec.id === val);
-                this._addItemToTable(record.id, record.display_name);
-            });
+            if (this.isCustom) {
+                this.addItemButton = document.createElement('we-button');
+                this.addItemButton.textContent = this.addItemTitle;
+                this.addItemButton.classList.add('o_we_list_add_optional');
+            } else {
+                // TODO use a real select widget ?
+                this.addItemButton = document.createElement('we-select');
+                this.addItemButton.classList.add('o_we_user_value_widget', 'o_we_add_list_item');
+                const divEl = document.createElement('div');
+                this.addItemButton.appendChild(divEl);
+                const togglerEl = document.createElement('we-toggler');
+                togglerEl.textContent = this.addItemTitle;
+                divEl.appendChild(togglerEl);
+                this.selectMenuEl = document.createElement('we-selection-items');
+                divEl.appendChild(this.selectMenuEl);
+            }
+            this.containerEl.appendChild(this.addItemButton);
+        }
+        currentValues.forEach(value => {
+            if (typeof value === 'object') {
+                this._addItemToTable(value.id, value.display_name);
+            } else {
+                this._addItemToTable(value, value);
+            }
+        });
+        if (!this.createWidget && !this.isCustom) {
             this._reloadSelectDropdown(currentValues);
         }
-        this.containerEl.appendChild(this.addItemButton);
         this._makeListItemsSortable();
     },
+    /**
+     * @override
+     */
+    getValue(methodName) {
+        if (this.createWidget && this.createWidget.getMethodsNames().includes(methodName)) {
+            return this.createWidget.getValue(methodName);
+        }
```

Comment:
I don't understand this, can you explain a bit?

Reply:
Basically when we click the create button, the `userValueNotification` triggers up to the `snippetOption` that will then do a `widget.getValue` on the widget when calling `_select`, in that case we want to delegate the `getValue` call to the `createWidget` so that it returns its value instead of returning the `listWidget`'s value which is a list of records.

Reply:
That's what I don't get:
1. is not `createWidget` always truthy since created in willStart? So feels like not only "when we click the create button".
2. getValue of a List widget... not returning a list of records? Feels like the create widget role should only be adding new elements to that list of records (then let the list widget notify its whole list)

Reply:
No, createWidget is always created in the m2m widget, and only passed to the list as an option when the list is achild widget of the m2m, which is "adopted" in the willstart, standalone ListUserValueWidgets don't have a create widget (though my original idea was to convert the current "add item" button into a different createWidget but I  gave up on that idea because it's a lot of work with the current system with little benefit)

getValue takes a `methodName` as argument, if the purpose is not to allow widget to return different values for different methods then I don't understand why this parameter is present at all. 

There is also the fact that record creation is non-trivial. For example, currently, blog uses custom prefixed IDs so that it can know which tags are new when saving the page, but other use cases may want to use different schemes to create the record.

Reply:
> getValue takes a methodName as argument, if the purpose is not to allow widget to return different values for different methods then I don't understand why this parameter is present at all.

Different **values** indeed, but still related to the type of value the widget exists for (here: a list of records). Example of why this parameter exists:
```xml
<we-range data-select-class="a|b|c" data-set-value=""/>
```
-> This creates a slider with 3 positions. The selectClass method receives a, b or c. The setValue method receives 0, 1 or 2. The state of the widget is a slider position, but the associated method value depends on the method.

> There is also the fact that record creation is non-trivial. For example, currently, blog uses custom prefixed IDs so that it can know which tags are new when saving the page, but other use cases may want to use different schemes to create the record.

Actually this is the same point as the previous one: I think the scheme to create the record should not defer from case to case. The creation should be totally transparent to the option and left to the widget, and that widget should notify the created values as part of the current values in a unified format that option can understand and rely on. The created records could also be mentioned in a parameter of the "params" argument which is given to the options methods.

But I understand this could require more refactoring and be non-trivial so I am ok to move forward with this, I don't really have the time to investigate in detail, so if it works for the current cases, that's what's important.

Reply:
Ok I understand your point. As you're saying it's a pretty big refactor to change this and since other tasks depend on this let's move forward with the current state, and hopefully when we migrate these to OWL this entire conversation will be pointless since passing callbacks from options to UVWs will be trivial and there will be no need to infer them from data-attribute names. :crossed_fingers: